R/utils.R

```diff
@@ -229,6 +242,30 @@ adjustToViewport <- function(x, viewport) {
   return(x)
 }
 
+#' Filter Data Table to Geolocation Viewport
+#' 
+#'
+#' This function will filter a data.table to the specified 
+#' latitude and longitude values.
+#' @param .dt data.table with columns for latitude and longitude
+#' @param latitude Character vector specifying name of latitude column
+#' @param longitude Character vector specifying name of longitude column
+#' @param viewport List of lists specifying the geolocation viewport
+#' @return data.table of filtered values 
+#' @export
+filterToGeolocationViewport <- function(.dt, latitude, longitude, viewport) {
+  if (is.null(viewport)) { return(.dt) }  
+
+  .dt <- .dt[.dt[[latitude]] >= viewport$latitude$xMin & .dt[[latitude]] <= viewport$latitude$xMax]
+  if (viewport$longitude$left < viewport$longitude$right) {
```

Comment:
Note: here we keep data on the viewport boundaries (with <= and => ) but in the [data service](https://github.com/VEuPathDB/EdaDataService/blob/f1883ce9e71c5bbdaedeb5e9e42007424fa298cf/src/main/java/org/veupathdb/service/eda/ds/plugin/pass/MapPlugin.java#L110-L115) we do not

Reply:
the logic is reversed bc one is inclusive and the other exclusive. result should be the same.